### ISSUE TITLE
Segmentation fault when running allennlp predict or evaluate with a saved model

### ISSUE THREAD

**Issue by ehsankddm** [allenai/allennlp#3312]
Allennlp predict and evaluate scripts give 'Segmentation fault' for a model produced by the allennlp train script and on an input file on which the model was trained originally.
The pastebin link for the config file, (toy) training and validation input files are provided below.

**To Reproduce**
Steps to reproduce the behavior
0. download files: [train.tsv](https://pastebin.com/Q28QmirM) [valid.tsv](https://pastebin.com/vyxJh9cK)  [config.json](https://pastebin.com/LdJ2jgpz)
1. allennlp train config.json -s output (download the train.tsv and valid.tsv from the links above and modify the paths in the config file)
2. allennlp predict output/model.tar.gz valid.tsv --output-file res.txt --cuda-device 0 --use-dataset-reader --predictor seq2seq
3. I get: 
> 2019-10-01 14:16:12,357 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.tokenizers.pretrained_transformer_tokenizer.PretrainedTransformerTokenizer'> from params {'do_lowercase': False, 'model_name': 'bert-base-cased'} and extras set()
> 2019-10-01 14:16:12,357 - INFO - allennlp.common.params - dataset_reader.source_tokenizer.model_name = bert-base-cased
> 2019-10-01 14:16:12,357 - INFO - allennlp.common.params - dataset_reader.source_tokenizer.do_lowercase = False
> 2019-10-01 14:16:12,357 - INFO - allennlp.common.params - dataset_reader.source_tokenizer.start_tokens = None
> 2019-10-01 14:16:12,357 - INFO - allennlp.common.params - dataset_reader.source_tokenizer.end_tokens = None
> 2019-10-01 14:16:12,679 - INFO - pytorch_transformers.tokenization_utils - loading file https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-cased-vocab.txt from cache at /home/khoddammohammadie/.cache/torch/pytorch_transformers/5e8a2b4893d13790ed4150ca1906be5f7a03d6c4ddf62296c383f6db42814db2.e13dbb970cb325137104fb2e5f36fe865f27746c6b526f6352861b1980eb80b1
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.tokenizers.tokenizer.Tokenizer'> from params {'type': 'word', 'word_splitter': {'do_lower_case': False, 'type': 'bert-basic'}} and extras set()
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.type = word
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.tokenizers.word_tokenizer.WordTokenizer'> from params {'word_splitter': {'do_lower_case': False, 'type': 'bert-basic'}} and extras set()
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.tokenizers.word_splitter.WordSplitter'> from params {'do_lower_case': False, 'type': 'bert-basic'} and extras set()
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.word_splitter.type = bert-basic
> 2019-10-01 14:16:12,790 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.tokenizers.word_splitter.BertBasicWordSplitter'> from params {'do_lower_case': False} and extras set()
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.word_splitter.do_lower_case = False
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.word_splitter.never_split = None
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.start_tokens = None
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.target_tokenizer.end_tokens = None
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.token_indexers.token_indexer.TokenIndexer'> from params {'do_lowercase': False, 'model_name': 'bert-base-cased', 'type': 'pretrained_transformer'} and extras set()
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.source_token_indexers.bert.type = pretrained_transformer
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.token_indexers.pretrained_transformer_indexer.PretrainedTransformerIndexer'> from params {'do_lowercase': False, 'model_name': 'bert-base-cased'} and extras set()
> 2019-10-01 14:16:12,791 - INFO - allennlp.common.params - dataset_reader.source_token_indexers.bert.model_name = bert-base-cased
> 2019-10-01 14:16:12,792 - INFO - allennlp.common.params - dataset_reader.source_token_indexers.bert.do_lowercase = False
> 2019-10-01 14:16:12,792 - INFO - allennlp.common.params - dataset_reader.source_token_indexers.bert.namespace = tags
> 2019-10-01 14:16:12,792 - INFO - allennlp.common.params - dataset_reader.source_token_indexers.bert.token_min_padding_length = 0
> 2019-10-01 14:16:13,107 - INFO - pytorch_transformers.tokenization_utils - loading file https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-cased-vocab.txt from cache at /home/khoddammohammadie/.cache/torch/pytorch_transformers/5e8a2b4893d13790ed4150ca1906be5f7a03d6c4ddf62296c383f6db42814db2.e13dbb970cb325137104fb2e5f36fe865f27746c6b526f6352861b1980eb80b1
> 2019-10-01 14:16:13,135 - INFO - allennlp.data.token_indexers.pretrained_transformer_indexer - Using token indexer padding value of 0
> 2019-10-01 14:16:13,135 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.token_indexers.token_indexer.TokenIndexer'> from params {'namespace': 'target_tokens', 'type': 'single_id'} and extras set()
> 2019-10-01 14:16:13,135 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.type = single_id
> 2019-10-01 14:16:13,135 - INFO - allennlp.common.from_params - instantiating class <class 'allennlp.data.token_indexers.single_id_token_indexer.SingleIdTokenIndexer'> from params {'namespace': 'target_tokens'} and extras set()
> 2019-10-01 14:16:13,135 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.namespace = target_tokens
> 2019-10-01 14:16:13,135 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.lowercase_tokens = False
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.start_tokens = None
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.end_tokens = None
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.target_token_indexers.tokens.token_min_padding_length = 0
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.source_add_start_token = True
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.delimiter = 
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.source_max_tokens = None
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.target_max_tokens = None
> 2019-10-01 14:16:13,136 - INFO - allennlp.common.params - dataset_reader.lazy = False
> 0it [00:00, ?it/s]2019-10-01 14:16:13,138 - INFO - allennlp.data.dataset_readers.seq2seq - Reading instances from lines in file at: valid.tsv
> 40it [00:00, 4395.62it/s]
> Segmentation fault

** System **

    OS: Amazon Linux AMI
    Python version: 3.7.4 (conda 4.7.12)
    AllenNLP version: from master
    PyTorch version: 1.2.0

**Comment by kernelmachine** [comment]
My guess is that there's some memory issue going on here, but it's hard to tell from the information you've provided. Can you confirm that the machine your running on has enough RAM to handle the BERT model you're evaluating/predicting with?

**Comment by ehsankddm** [comment]
Thanks, I checked it and none of the resources are saturated.

**Comment by ehsankddm** [comment]
The problem doesn't occur with a clean installation from scratch. It happened after pulling the latest commits from the master branch.
